### PR TITLE
Replaced liveTabs with flowTabs.

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/414730916066338/1201550925053699/f

### Description
Removed the usage of `liveTabs` and replaced them with `flowTabs`.

### Steps to test this PR

- install from this branch
- open a 8-9 tabs using different methods: search something / open link in background tab / open in new tab
- switch between tabs
- close tabs from tab switcher activity
- open new tabs from tab switcher activity
- close all tabs from tab switcher activity
- open tabs and use the fire button
- check there is no unusual behaviour

https://user-images.githubusercontent.com/7963079/154678444-54cbe293-9d90-41c9-8e72-301afec741d1.mp4

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt [hunk 1]
```diff
@@ -402,7 +402,7 @@ class CtaViewModel @Inject constructor(
         }
     }
 
-    private fun forceStopFireButtonPulseAnimationFlow() = tabRepository.flowTabs.distinctUntilChanged()
+    private fun forceStopFireButtonPulseAnimationFlow() = tabRepository.flowTabs
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt:405]: Removed `.distinctUntilChanged()` as it's done in the repository.

File: app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt [hunk 2]
```diff
@@ -271,25 +272,18 @@ class TabSwitcherActivity : DuckDuckGoActivity(), TabSwitcherListener, Coroutine
...
     override fun onDestroy() {
         super.onDestroy()
-        viewModel.deletableTabs.removeObservers(this)
         // we don't want to purge during device rotation
         if (isFinishing) {
             launch { viewModel.purgeDeletableTabs() }
         }
     }
 
-    private fun clearObserversEarlyToStopViewUpdates() {
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt:288]: We don't need to manually cleanup anymore.

File: app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt [hunk 3]
```diff
@@ -51,9 +51,7 @@ class TabDataRepository @Inject constructor(
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope
 ) : TabRepository {
 
-    override val liveTabs: LiveData<List<TabEntity>> = tabsDao.liveTabs()
-
-    override val flowTabs: Flow<List<TabEntity>> = tabsDao.flowTabs()
+    override val flowTabs: Flow<List<TabEntity>> = tabsDao.flowTabs().distinctUntilChanged()
```
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt:54]: I think livedata does not behave like flow in all cases. Is there anything special we'd need to test

Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt:54]: Nothing very specific apart from what's in the description.

File: app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt (hunk 2 above)
Comment (joshliebe) [reply app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherActivity.kt:288]: Love this! 🤩

**Review by joshliebe — APPROVED** [review]
This is great @anikiki! Didn't see any issues while testing ✅

**Pull request merged**